src/optimizing/Analyze_rule.mli

```diff
@@ -25,7 +25,10 @@ type prefilter = Semgrep_prefilter_t.formula * (string -> bool)
  *
  * Note that this function use Common.memoized on the rule id
  *)
-val regexp_prefilter_of_rule : Rule.t -> prefilter option
+val regexp_prefilter_of_rule :
+  hmemo:(Rule_ID.t, prefilter option) Hashtbl.t option ->
```

Comment:
like I said before, better to define a type for this argument, with a good explanation in comment for its need.